app/components/Edit/EditSchedule.js

```diff
@@ -1,151 +1,192 @@
-import { createTemplateSchedule } from '../../utils/index';
+import { timeToTimeInputValue, stringToTime, daysOfTheWeek } from '../../utils/index';
+import EditScheduleDay from './EditScheduleDay';
 
-class EditSchedule extends Component {
-  constructor(props) {
-    super(props);
-
-    let scheduleMap = {
-      Monday: {
-        opens_at: null,
-        day: 'Monday',
-        closes_at: null,
-        id: 1,
-      },
-      Tuesday: {
-        opens_at: null,
-        day: 'Tuesday',
-        closes_at: null,
-        id: 2,
-      },
-      Wednesday: {
-        opens_at: null,
-        day: 'Wednesday',
-        closes_at: null,
-        id: 3,
-      },
-      Thursday: {
-        opens_at: null,
-        day: 'Thursday',
-        closes_at: null,
-        id: 4,
-      },
-      Friday: {
-        opens_at: null,
-        day: 'Friday',
-        closes_at: null,
-        id: 5,
-      },
-      Saturday: {
-        opens_at: null,
-        day: 'Saturday',
-        closes_at: null,
-        id: 6,
-      },
-      Sunday: {
-        opens_at: null,
-        day: 'Sunday',
-        closes_at: null,
-        id: 7,
-      },
-    };
-    props.schedule && props.schedule.schedule_days.forEach(function(day) {
-      scheduleMap[day.day] = day;
+function buildSchedule(schedule) {
+  let scheduleId = schedule.id;
+  let tempSchedule = {
+    Monday: [{ opens_at: null, closes_at: null, scheduleId }],
+    Tuesday: [{ opens_at: null, closes_at: null, scheduleId }],
+    Wednesday: [{ opens_at: null, closes_at: null, scheduleId }],
+    Thursday: [{ opens_at: null, closes_at: null, scheduleId }],
+    Friday: [{ opens_at: null, closes_at: null, scheduleId }],
+    Saturday: [{ opens_at: null, closes_at: null, scheduleId }],
+    Sunday: [{ opens_at: null, closes_at: null, scheduleId }],
+  };
+  if (schedule) {
+    schedule.schedule_days.forEach((curr) => {
+      curr.openChanged=false;
+      curr.closeChanged=false;
+      tempSchedule[curr.day].unshift(curr);
+    });
+
+    let tempDaySched = [];
+    Object.keys(tempSchedule).forEach((day) => {
+      tempDaySched = tempSchedule[day];
+      if (tempDaySched.length > 1) {
+        tempDaySched.pop();
+      }
     });
+  }
+  return tempSchedule;
+}
+
+class EditSchedule extends Component {
+constructor(props) {
+  super(props);
+
+  let scheduleMap = {};
+  props.schedule && props.schedule.schedule_days.forEach(function(day) {
+    scheduleMap[day.day] = day;
+  });
+
+  this.state = {
+    scheduleMap: scheduleMap,
+    schedule_days: {},
+    scheduleId: props.schedule ? props.schedule.id : null,
+    scheduleDays: buildSchedule(props.schedule),
+    uuid: -1,
+    scheduleId: this.props.schedule ? this.props.schedule.id : null,
+    open24Hours: {
+      "Monday": false,
+      "Tuesday": false,
+      "Wednesday": false,
+      "Thursday": false,
+      "Friday": false,
+      "Saturday": false,
+      "Sunday": false,
+    }
```

Comment:
I'm a little worried about this `open24Hours` property because it seems redundant with `scheduleDays`. What happens if we let the two get out of sync and they contradict each other? Shouldn't `open24Hours` be computed from `scheduleDays`, which avoids the denormalization issue?

Reply:
I removed all the code related to 24 hours since that is going to be implemented later